webapp/sources/rudder/rudder-core/src/main/scala/com/normation/cfclerk/services/impl/GitTechniqueReader.scala

```diff
@@ -379,7 +381,10 @@ class GitTechniqueReader(
 
     val managed = Managed.make(
       for {
-        currentId <- revisionProvider.currentRevTreeId
+        currentId <- revId match {
+                       case None    => revisionProvider.currentRevTreeId
+                       case Some(r) => GitFindUtils.findRevTreeFromRevString(repo.db, r.value)
```

Comment:
I find this one really scary - i'm unsure of the perf of this one, and maybe we should consider a cache somewhere

Reply:
yes, this is a PoC, and I don't want to address potential perf problem until I understand the whole design space and what need to be done. 
I think we will need to completly change the way we use ldap, even remove it and replace it by a pur cache for `(id,revId)`. But it's too early to see the whole scope of change, even more how to iterate on them.

Reply:
:+1: 
If you need to test that, you can create a bash script that:
creates many non empty files in git
git add these files & git commit with a random message
git remove these files & git commit with a random message
and start again

It makes a serious dent in perf of git after several thousands commit

Reply:
it need to make `git gc`, and performances come back to baseline

Reply:
actually, perhap not (but yes, we need to test that on big git, and also find what is the best fashion to get what we need from it)